app/backend/approaches/chatreadretrieveread.py

```diff
@@ -77,13 +79,22 @@ def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]) -> A
             self.chatgpt_token_limit - len(user_q)
             )
 
-        chat_completion = openai.ChatCompletion.create(
-            deployment_id=self.chatgpt_deployment,
-            model=self.chatgpt_model,
-            messages=messages,
-            temperature=0.0,
-            max_tokens=32,
-            n=1)
+        if self.openai_type == "azure":
+            chat_completion = openai.ChatCompletion.create(
+                deployment_id=self.chatgpt_deployment,
```

Comment:
@carlotta94c I'm wondering if we can avoid all the if/else with the OpenAI calls. Is it possible to just pass None or an empty string here, or does that completely confuse the OpenAI SDK?

Reply:
Suggestion from Krista:

kwargs = {"model": self.chatgpt_model} if api_type == "openai" \
    else {"deployment_id": self.chatgpt_deployment}

Can be used like:

response = openai.ChatCompletion.create(messages=messages, stream=True, **kwargs)

Can you try that style? That should reduce if/else

Reply:
I applied Krista's suggestion and it works well. Can you please check it @pamelafox ?